src/undomanager.js

```diff
@@ -22,16 +22,21 @@ define([
         $('.fd-undo-container').append(UNDO_ALERT);
     }
 
-    function toggleAlert(undoStack) {
+    function toggleAlert(undoStack, vellum) {
```

Comment:
Another option would be to have `UndoManager` publish events (see [how](https://github.com/dimagi/Vellum/blob/deletion-panel-resize/src/form.js#L140) the form [does it](https://github.com/dimagi/Vellum/blob/deletion-panel-resize/src/form.js#L178-L181)), which would allow decoupling of UI code from `UndoManager`. The form could then hook up the relevant event(s) to call `vellum.adjustToWindow()` after creating the undo manager. Unfortunately the form is fairly tightly coupled with the UI at this point; ideally the form would not be interacting directly with the UI either since it is more of a model class.

If you want to dive deeper and do dependency injection, you could pass a fully configured and event-hooked undo manager into the form constructor, further decoupling the form from UI code. If you go this (DI) route, you might consider doing it in a separate PR. Currently the form instance is [created in parser.js](https://github.com/dimagi/Vellum/blob/deletion-panel-resize/src/parser.js#L35), which, while part of the vellum core, seems like a less than ideal place to be connecting model events with UI callbacks.